extensions/resteasy-reactive/quarkus-resteasy-reactive/deployment/src/main/java/io/quarkus/resteasy/reactive/server/deployment/devconsole/StaticResourceInfo.java

```diff
@@ -0,0 +1,29 @@
+package io.quarkus.resteasy.reactive.server.deployment.devconsole;
+
+import java.nio.file.Path;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+import java.util.TreeMap;
+
+public class StaticResourceInfo {
+
+    Map<Path, StaticFile> resourceMap = new TreeMap<>();
+
+    public static final class StaticFile {
+        public String lastName;
+        public boolean isFolder;
+        public List<StaticFile> children = new ArrayList<>();
+
+        public StaticFile(String lastName, boolean isFolder, List<StaticFile> children) {
+            this.lastName = lastName;
+            this.isFolder = isFolder;
+            this.children = children;
```

Comment:
If you initialise `children` with `new ArrayList<>()` then change it in the constructor you will have created the array for nothing. I think you can drop the `children` parameter in the constructor and keep the one you create in your field init.

Reply:
ok :)